src/main/java/org/openstreetmap/atlas/geography/atlas/change/ChangeAtlas.java

```diff
@@ -120,61 +139,65 @@ public Node node(final long identifier)
     @Override
     public long numberOfAreas()
     {
-        throw new UnsupportedOperationException();
+        return Iterables.size(areas());
```

Comment:
It might be worth caching these values once you've computed them with `Iterables.size()`? What do you think? Maybe it's just unnecessary code bloat?

Reply:
```java
private Optional<Long> numberOfAreas = Optional.empty();
.
.
.
public long numberOfAreas()
{
    if (numberOfAreas.isPresent()) {
         return numberOfAreas.get();
    } else {
        numberOfAreas = Optional.of(Iterables.size(areas()));
        return numberOfAreas.get();
    }
}
```

Reply:
Good idea.